noir-projects/noir-contracts/contracts/token_with_refunds_contract/src/types/token_note.nr

```diff
@@ -215,44 +215,44 @@ impl PrivatelyRefundable for TokenNote {
 
         // 3. We do the necessary conversion for values relevant for the sponsored user point.
         // We use the unsafe version because the multi_scalar_mul will constrain the scalars.
-        let funded_amount_scalar = from_field_unsafe(funded_amount);
         let user_npk_m_hash_scalar = from_field_unsafe(user_npk_m_hash);
         let user_randomness_scalar = from_field_unsafe(user_randomness);
 
-        // 4. We compute `G_amt * funded_amount + G_npk * user_npk_m_hash + G_rnd * randomness`.
+        // 4. We compute `G_npk * user_npk_m_hash + G_rnd * randomness`.
         let incomplete_user_point = multi_scalar_mul(
-            [G_amt, G_npk, G_rnd],
-            [funded_amount_scalar, user_npk_m_hash_scalar, user_randomness_scalar]
+            [G_npk, G_rnd],
+            [user_npk_m_hash_scalar, user_randomness_scalar]
         );
 
         // 5. At last we return the points.
         (incomplete_fee_payer_point, incomplete_user_point)
     }
 
-    fn complete_refund(incomplete_fee_payer_point: Point, incomplete_user_point: Point, transaction_fee: Field) -> (Field, Field) {
-        // 1. We convert the transaction fee to high and low limbs to be able to use BB API.
+    fn complete_refund(incomplete_fee_payer_point: Point, incomplete_user_point: Point, funded_amount: Field, transaction_fee: Field) -> (Field, Field) {
+        // 1. We check that user funded the fee payer contract with at least the transaction fee.
+        assert(!funded_amount.lt(transaction_fee), "tx fee is higher than funded amount"); // funded_amout >= transaction_fee
+
+        // 2. We convert the transaction fee and refund amount to high and low limbs to be able to use BB API.
         // We use the unsafe version because the multi_scalar_mul will constrain the scalars.
         let transaction_fee_scalar = from_field_unsafe(transaction_fee);
+        let refund_scalar = from_field_unsafe(funded_amount - transaction_fee);
 
-        // 2. We compute the fee point as `G_amt * transaction_fee`
-        let fee_point = multi_scalar_mul(
-            [G_amt],
-            [transaction_fee_scalar]
-        );
+        // 3. We compute the fee point as `G_amt * transaction_fee`
+        let fee_point = multi_scalar_mul([G_amt], [transaction_fee_scalar]);
 
-        // 3. Now we leverage homomorphism to privately add the fee to fee payer point and subtract it from
-        // the sponsored user point in public.
-        let fee_payer_point = incomplete_fee_payer_point + fee_point;
-        let user_point = incomplete_user_point - fee_point;
+        // 4. We compute the refund point as `G_amt * refund`
+        let refund_point = multi_scalar_mul([G_amt], [refund_scalar]);
 
-        assert_eq(user_point.is_infinite, false);
```

Comment:
Nuked this assert because:
<img width="608" alt="image" src="https://github.com/user-attachments/assets/0fc7c4e3-3f7f-45a3-81b4-c6173880957b">